src/client.ts

```diff
@@ -87,6 +99,35 @@ export class APIClient {
       body,
     });
 
+    // According to BambooHR documentation, a 429 error is a limit exceeded
+    // regarding adding employees.  A 503 is a "currently unavailable" 
+    // error that is most commonly due to rate limiting.  It may contain
+    // a 'Retry-After' value in the header specifying how long to wait.
+    // Unfortunately, we don't have documenation or examples for this, and
+    // the HTTP spec states ths can either be a second value or an HTTP-date.
+    // For now, we're checking if the value is a number and attempting to 
+    // handle either option with a max upper limit for how long we'll wait
+    // to avoid worst case scenarios for failed parsing.
+    if(response.status === 503) {
```

Comment:
nit: consider pulling out this block of code into a separate function for readability.

Reply:
Updated in the latest commit.